components/script/dom/window.rs

```diff
@@ -277,6 +280,9 @@ pub struct Window {
 
     /// Timers used by the Console API.
     console_timers: TimerSet,
+
+    /// All the MediaQueryLists we need to update
+    media_query_lists: DOMRefCell<Vec<JS<MediaQueryList>>>,
```

Comment:
We can't store WeakRefVec directly because it has not JSTraceable implementation (and seemingly can't have one). I did what Range does and made a WeakMediaQueryListVec type which wraps WeakVec.